src/NuGet.Clients/NuGet.VisualStudio.Common/EnvDTEProjectInfoUtility.cs

```diff
@@ -190,6 +190,7 @@ public static EnvDTE.Project GetActiveProject(IVsMonitorSelection vsMonitorSelec
 
         private static T GetPropertyValue<T>(EnvDTE.Project envDTEProject, string propertyName)
         {
+            ThreadHelper.ThrowIfNotOnUIThread();
```

Comment:
Do we need to add this check for `private` and `internal` methods where we have control over the callers? As per [`VSTHRD010 doc`](https://github.com/Microsoft/vs-threading/blob/master/doc/analyzers/VSTHRD010.md) ` For example, if method A() invokes a type that we know from configuration requires the main thread, and B() calls A(), then the B method also needs the UI thread transitively. This analyzer flags B() as needing to call a method that throws if not already on the main thread only when A() is written to call such a method.`

This comment is applicable to all the `private` methods in the PR.

Reply:
That comment doesn't say anything about not adding threading checks on private methods. 
It says that the analyzer will flag the pattern. 

Am I misreading it?